Pod/Classes/WPVideoPlayerView.m

```diff
@@ -41,18 +45,54 @@ - (void)commonInit {
     self.player = [[AVPlayer alloc] init];
     self.playerLayer = [AVPlayerLayer playerLayerWithPlayer: self.player];
     [self.layer addSublayer: self.playerLayer];
+    [self addSubview:self.controlToolbar];
+
+    __weak __typeof__(self) weakSelf = self;
+    self.timeObserver = [self.player addPeriodicTimeObserverForInterval:CMTimeMakeWithSeconds(1, NSEC_PER_SEC) queue:nil usingBlock:^(CMTime time) {
+        [weakSelf updateVideoDuration];
+    }];
 }
 
 - (void)dealloc {
-    [self.playerItem removeObserver:self forKeyPath: @"status"];
+    [_playerItem removeObserver:self forKeyPath: @"status"];
     [[NSNotificationCenter defaultCenter] removeObserver: self];
-    [self.player pause];
+    [_player removeTimeObserver:self.timeObserver];
+    [_player pause];
     _asset = nil;
+    _player = nil;
 }
 
 - (void)layoutSubviews {
     [super layoutSubviews];
     self.playerLayer.frame = self.bounds;
+    CGFloat position = toolbarHeight;
```

Comment:
Could just do this as a ternary? `CGFloat position = (self.controlToolbarHidden) ? 0 : toolbarHeight;`